library/psa_crypto.c

```diff
@@ -7449,12 +7449,25 @@ psa_status_t mbedtls_psa_crypto_configure_entropy_sources(
     void (* entropy_init)(mbedtls_entropy_context *ctx),
     void (* entropy_free)(mbedtls_entropy_context *ctx))
 {
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+
+#if defined(MBEDTLS_THREADING_C)
+    mbedtls_mutex_lock(&mbedtls_threading_psa_rngdata_mutex);
```

Comment:
Why do we use the RNG mutex for this? The reason this function needs to take a mutex is because it accesses members of the global data structure and that structure is protected by the global data mutex.

Reply:
Thanks for explaining it off-line: because `global_data.rng_state` is protected by the RNG mutex.